homeassistant/components/bsblan/config_flow.py

```diff
@@ -41,6 +45,53 @@ async def async_step_user(
         self.username = user_input.get(CONF_USERNAME)
         self.password = user_input.get(CONF_PASSWORD)
 
+        return await self._validate_and_create()
+
+    async def async_step_zeroconf(
+        self, discovery_info: zeroconf.ZeroconfServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle Zeroconf discovery."""
+
+        self.host = str(discovery_info.ip_address)
+        self.port = discovery_info.port or DEFAULT_PORT
+
+        # Check if the device is already configured
+        for entry in self._async_current_entries():
+            if (
+                entry.data.get(CONF_HOST) == self.host
+                and entry.data.get(CONF_PORT) == self.port
+            ):
+                return self.async_abort(reason="already_configured")
```

Comment:
We should update the config entry with the unique id.

You already have this code in the creation function of the entry

Reply:
Extra note here: configuring the same device via the user flow should abort. If they want to set the IP manually because zeroconf doesn't work for example, they should use a reconfigure flow